### PR TITLE
feat: Add support for docker-compose v2

### PR DESCRIPTION
**Pull request by sjinks**
This PR adds support for docker-compose 2.x.

The main difference between v1 and v2 is that v2 uses a [different word separator on container names](https://docs.docker.com/compose/cli-command-compatibility/).

### CODE REVIEW

File: plugins/lando-events/lib/utils.js [hunk 1]
```diff
@@ -41,9 +41,10 @@ exports.events2Runz = (cmds, app, data = {}) => _.map(cmds, cmd => {
   if (app.services && !_.includes(app.services, service)) {
     throw new Error(`This app has no service called ${service}`);
   }
+  const separator = app._config.composeSeparator;
```
Comment (sjinks) [inline plugins/lando-events/lib/utils.js:44]: I know this is not an ideal solution, but I can't seem to find another way to get Lando config when I have only App :-(

**Comment by sjinks** [comment]
There is a test branch with these changes: https://github.com/Automattic/vip-cli/tree/migrate/lando

**Comment by reynoldsalec** [comment]
Know this is a bit stale, but considering the impending removal of Docker Compose v1 from Docker Desktop, I've been more interested in how easy it would be to move Lando to Docker Compose v2.

So far everything I've seen seems to be related to the separator. Docker Compose v2 does introduce more feedback that shows up in the CLI and is arguably "distracting" from the Lando experience...not sure if we care about that.

I think the main problem is the way this version of the code tries to detect the Docker Compose version being used: https://github.com/lando/core/pull/19/files#diff-3b5eaf141d60a6350440399209dc8bd2ce83101f3416b56a1cdc165f39e6f2a0R116

This code relies on Lando's method of retrieving the Compose Version from the MacOS `Info.plist`: https://github.com/lando/core/blob/main/lib/daemon.js#L182 However, I think this method no longer returns a result (manually checking that file seems to confirm that it doesn't contain the expected values any more), and even if it did, the intent of that code seems to be to just to get the version of Docker Compose v1 packaged with Docker Desktop.

If we were going to implement something like this code, we'd want to decide whether we're offering backwards compatibility for Docker Compose V1. If we do want to do that, we'd probably want to set some sort of toggle for the appropriate version of Docker Compose that can be user configured on a global/app basis and defaults to a "sane" value (probably whatever version of Compose is used when `docker-compose` is executed on the user's native shell).

Making a hard shift to Docker Compose v2 seems possible. That would entail...

1. Existing apps needing to be rebuilt
2. Linux users [running from source](https://docs.lando.dev/getting-started/installation.html#from-source) needing to reconfigure their setup to use Docker Compose v2

**Comment by reynoldsalec** [comment]
One note: I did confirm that volumes are preserved between rebuilds when rebuilding an app originally run with Compose V1. Service names with underscores seem to work fine in Docker Compose V2.

**Comment by reynoldsalec** [comment]
@pirog had some ideas on how we might implement this in Lando v3 that would help "prototype" the way Lando v4 would use Docker Compose, as well as handle other orchestrator (Kubernetes comes to mind).

Currently [lib/engine.js](https://github.com/lando/core/blob/main/lib/engine.js) runs both Docker and Docker Compose commands. [lib/router.js](https://github.com/lando/core/blob/main/lib/router.js) is in charge of routing those to the appropriate component (ie Docker or Docker Compose), with [lib/compose.js](https://github.com/lando/core/blob/main/lib/compose.js) executing the Docker Compose commands.

In Lando v4 we'll have better handling of the "routing" to components (IE replacing the translation that occurs in engine/router). To prepare for that, we could transform compose.js into a class that contains helpers methods for translating separators, making sure that container names are appropriate for the orchestrator, finding its installation location, version, etc. We'd implement that class for Docker Compose V1, making sure it passes all our testing, then create an additional implementation for V2.

To use these two classes in Lando 3, we'd probably want a toggle both on a global level (perhaps in `config.yml`) and on an app-level (in the Landofile) to toggle between the orchestrator options.

**Comment by pirog** [comment]
@sjinks we are giving this another look using your code over here https://github.com/lando/core/pull/52

**Pull request closed without merging**